parkbenchcommon/broadcaster.py

```diff
@@ -0,0 +1,138 @@
+        self.broadcast_name = broadcast_name
+
+        tmpfs_path = os.path.join(broadcast_path, program_name)
+        tmpfs.mount_tmpfs(tmpfs_path, TMPFS_SIZE)
+
+        self._create_directory(self.broadcast_path)
+        # Set permissions to xrw-------
+        group_rw_mode = stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR | stat.S_IXGRP | stat.S_IRGRP
+        self._set_file_permissions(self.broadcast_path, group_rw_mode, uid, gid)
+
+        self.logger.info("Broadcaster for eacon %s from program %s initialized.",
+                         broadcast_name, program_name)
+
+    def issue(self):
+        """Place a new file in the broadcast directory. Will raise an exception if it fails."""
+        self.logger.info(
+            "Sending broadcast %s for program %s.", self.broadcast_name, self.program_name)
+        now = datetime.datetime.now().isoformat()
+        random_number = os.urandom(16).encode('hex')
+
+        broadcast_filename = '%s---%s' % (now, random_number)
+        final_path = os.path.join(self.broadcast_path, broadcast_filename)
+
+        try:
+            previous_broadcasts = os.listdir(self.broadcast_path)
+            open(final_path, 'a').close()
+
+            for broadcast_file in previous_broadcasts:
+                os.remove(os.path.join(self.broadcast_path, broadcast_file))
+
+        except Exception as exception:
+            message = \
+                'Could not create broadcast file for broadcast %s, program %s. %s:%s\n%s' \
+                % (self.broadcast_name, self.program_name, type(exception).__name__,
+                   str(exception), traceback.format_exc())
+            self.logger.critical(message)
+            # TODO: Implement exception chaining when we move to Python 3.
+            raise BroadcasterBroadcastException(message)
+
+    def _set_file_permissions(self, path, mode, uid, gid):
+        """Sets permissions for a file. Will raise an exception if it fails.
+
+        path: The file to modify.
+        mode: The mode to set.
+        uid: The owner to set.
+        gid: The group to set.
+        """
+        try:
+            os.chown(path, uid, gid)
+            os.chmod(path, mode)
+
+        except Exception as exception:
+            message = \
+                'Could not set permissions for file %s for broadcast %s for program %s. ' \
+                '%s: %s\n%s' % (path, self.broadcast_name, self.program_name,
+                                type(exception).__name__, str(exception),
+                                traceback.format_exc())
+            self.logger.error(message)
+            # TODO: Implement exception chaining when we move to Python 3.
+            raise BroadcasterInitException(message)
+
+    def _create_directory(self, path):
+        """Create a directory while handling errors. Will raise an exception if it fails.
+
+        path: The directory to create.
+        """
+        path = os.path.abspath(path)
+        if not os.path.isdir(path):
```

Comment:
This doesn't check if the directory is a file, although I guess that will be caught by the exception handler below.